Zapper-fi/studio
# feat(clever): implement vesting

## Description

Implemented vesting for Clever.

## Checklist

- [X] (optional) As a contributor, my Ethereum address/ENS is: 0xcfb3b99e0a7ad8008a5bd5d3255e338f493f8c15

## How to test?

The values returned from api and in the Clever ui should match.

![image](https://user-images.githubusercontent.com/20791144/218304008-37cd05e3-892e-4ad0-be36-860df08d875a.png)
should match the response (not exactly because of difference in timing.):
```
{
   "0xcfb3b99e0a7ad8008a5bd5d3255e338f493f8c15":{
      "products":[
         {
            "label":"Vesting",
            "assets":[
               {
                  "key":"147083998",
                  "type":"contract-position",
                  "appId":"clever",
                  "groupId":"vesting",
                  "network":"ethereum",
                  "address":"0x84c82d43f1cc64730849f3e389fe3f6d776f7a4e",
                  "tokens":[
                     {
                        "metaType":"vesting",
                        "type":"base-token",
                        "network":"ethereum",
                        "address":"0x72953a5c32413614d24c29c84a66ae4b59581bbf",
                        "symbol":"CLEV",
                        "decimals":18,
                        "price":23.97,
                        "balance":25.527939639673004,
                        "balanceRaw":"25527939639673002750",
                        "balanceUSD":611.9047131629619
                     },
                     {
                        "metaType":"claimable",
                        "type":"base-token",
                        "network":"ethereum",
                        "address":"0x72953a5c32413614d24c29c84a66ae4b59581bbf",
                        "symbol":"CLEV",
                        "decimals":18,
                        "price":23.97,
                        "balance":2.5013336303557057,
                        "balanceRaw":"2501333630355706073",
                        "balanceUSD":59.95696711962626
                     }
                  ],
                  "dataProps":{
                     
                  },
                  "displayProps":{
                     "label":"Vesting CLEV",
                     "images":[
                        "https://storage.googleapis.com/zapper-fi-assets/tokens/ethereum/0x72953a5c32413614d24c29c84a66ae4b59581bbf.png",
                        "https://storage.googleapis.com/zapper-fi-assets/tokens/ethereum/0x72953a5c32413614d24c29c84a66ae4b59581bbf.png"
                     ],
                     "statsItems":[
                        
                     ]
                  },
                  "balanceUSD":671.8616802825882
               }
            ],
            "meta":[
               
            ]
         }
      ],
      "meta":[
         {
            "label":"Total",
            "value":671.8616802825882,
            "type":"dollar"
         },
         {
            "label":"Assets",
            "value":671.8616802825882,
            "type":"dollar"
         },
         {
            "label":"Debt",
            "value":0,
            "type":"dollar"
         }
      ]
   }
}
```
